modyn/config/schema/pipeline/trigger/drift/detection_window/amount.py

```diff
@@ -0,0 +1,60 @@
+from __future__ import annotations
+
+from functools import cached_property
+from typing import Annotated, Literal
+
+from pydantic import Field
+
+from modyn.config.schema.base_model import ModynBaseModel
+from modyn.const.regex import REGEX_TIME_UNIT
+from modyn.utils.utils import SECONDS_PER_UNIT
+
+
+class _BaseWindowingStrategy(ModynBaseModel):
+    allow_overlap: bool = Field(
+        False,
+        description=(
+            "Whether the windows are allowed to overlap. This is useful for time-based windows."
```

Comment:
Why is it not useful for amount based windows? In my understanding this is relevant for the time after a trigger is performed, and we could have overlaps also in amount windows?

Reply:
You're right. We can also decide the overlap based on the sample IDs. When I first wrote this, I only had the sample times available in the offline exploration. The implementation already supports it :)